configuration/src/main/java/io/airlift/configuration/ConditionalModule.java

```diff
@@ -64,16 +65,28 @@ public static <T> Module conditionalModule(Class<T> config, Predicate<T> predica
 
     public static <T> Module conditionalModule(Class<T> config, String prefix, Predicate<T> predicate, Module module)
     {
-        return new ConditionalModule<>(config, Optional.of(prefix), predicate, module);
+        return new ConditionalModule<>(config, Optional.ofNullable(prefix), predicate, module);
     }
 
+    public static <T> Module conditionalModule(Key<T> key, Class<T> config, String prefix, Predicate<T> predicate, Module module)
+    {
+        return new ConditionalModule<>(key, config, Optional.ofNullable(prefix), predicate, module);
+    }
+
+    private final Key<T> key;
     private final Class<T> config;
     private final Optional<String> prefix;
     private final Predicate<T> predicate;
     private final Module module;
 
     private ConditionalModule(Class<T> config, Optional<String> prefix, Predicate<T> predicate, Module module)
     {
+        this(Key.get(config), config, prefix, predicate, module);
+    }
+
```

Comment:
This is a private constructor so we can just remove this and update the callers to use the new one.